cli/compilers/js.rs

```diff
@@ -0,0 +1,24 @@
+use crate::compilers::CompiledModule;
+use crate::compilers::CompiledModuleFuture;
+use crate::file_fetcher::SourceFile;
+use crate::state::ThreadSafeState;
+use std::str;
+
+pub struct JsCompiler {}
+
+impl JsCompiler {
```

Comment:
The idea is that in the future these will be trait impl ?

Reply:
Yes, exactly. I'll try to refactor TsCompiler not to require `state` arg during compilation and factor out `Compiler` trait

Reply:
This gonna need some more thought... We're updating metrics on `state` when spinning up TS compiler so ATM it looks like `state` is a required argument. I'm gonna leave it as is now and addresses in follow up PR.